### PR TITLE
docs: compile page layouts from pug

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Created a parser that converts Pug files into our current structure for pages located [here](https://github.com/vuetifyjs/vuetify/tree/master/packages/docs/src/data). This is setup to allow incremental updates to the docs and follows this pattern currently:

```pug
// headers must use the h2 tag
h2 vue-cli-header

// plain text
| vue-cli-text-1

// Alerts, value is the type
alert(value="warning") vue-cli-alert-1

// components receive values from attributes
component(value="...")

code(lang="bash" src="vue-add-capacitor")

// Image src goes in value, alt text in block
img(value="quick-start/vue_ui.png") ui-img-1
```

## How Has This Been Tested?
Compiled the output data file.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by sh7dm** [comment]
Oh, great! Maybe snippets also can be extracted? It'll be awesome to have smth like this:
```markdown
<!-- snippets have 3 parts -->
<!-- `{file type} {snippet name} {code}` -->
`'` bash vue-add-capacitor
ls /
`'`
```

**Comment by MajesticPotatoe** [comment]
> Oh, great! Maybe snippets also can be extracted?

I believe this is just to replace the json structure of data files. Snippet directory would still exist, and seems like it already parses / links the snippet file.

**Comment by johnleider** [comment]
> Oh, great! Maybe snippets also can be extracted? It'll be awesome to have smth like this:
> 
> ```gfm
> 
> 
> `'` bash vue-add-capacitor
> ls /
> `'`
> ```
> 
> then the compiler could extract and move code to snippets directory

We wouldn't want to do the snippets inline because they are reusable.

**Comment by KaelWD** [comment]
I don't like how it's still JSON, you've just created a new DSL on top of the existing one and it's even more limiting.

**Comment by johnleider** [comment]
@KaelWD I did it as such so that it's easy to start converting as opposed to having to change the underlying structure of how the docs are built. We can do that, but it's out of the scope of this.

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/core-contributors @vuetifyjs/core-team

File: packages/docs/build/pug-to-json.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    compileDebug: false,
+    filename: this.resourcePath,
+    filters: {
+      code: text => text
+    },
+    plugins: [{
+      preFilters (ast) {
+        const parents = []
+        walk(ast, node => {
+          if (!pageSection || ['h2', 'up-next', 'usage', 'usage-new', 'api', 'examples'].includes(node.name)) {
```
Comment (KaelWD) [inline packages/docs/build/pug-to-json.js]: Not sure about this bit, it seems like most components except markup and alerts are in a new section. Might be a better idea to be explicit about this?

File: packages/docs/build/pug-to-json.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            case 'Tag':
+              if (node.name === 'h2') {
+                pageSection.children.push({
+                  type: 'heading',
+                  lang: node.block.nodes[0].val
+                })
+              } else if (node.name === 'code') {
+                const lang = eval(node.attrs.find(attr => attr.name === 'lang').val)
+                const src = eval(node.attrs.find(attr => attr.name === 'src').val)
+                // TODO: inline code
```
Comment (KaelWD) [inline packages/docs/build/pug-to-json.js]: We could keep the edit button and even have it link to the correct line by using `node.line`

**Pull request merged**